File: dehaze-java/src/main/java/com/pei/dehaze/model/entity/SysAiMcpServer.java
package com.pei.dehaze.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.pei.dehaze.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 外部 MCP Server 注册表（表 sys_ai_mcp_server）。
 *
 * <p>credentials 为 AES 密文 JSON（{"api_key":"<密文>","extra":{"k":"<密文>"}}），
 * 仅录入/更新，不回显明文、不暴露给 LLM。
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class SysAiMcpServer extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String description;

    /** streamable-http / sse（stdio 无网络端点，不支持） */
    private String protocolType;

    private String endpoint;

    private String authType;

    private String credentials;

    /** online / offline */
    private String health;

    private LocalDateTime lastCheckTime;

    private Integer status;

    /** 冗余工具数量，避免列表逐条子查询 */
    private Integer toolCount;

    @TableLogic(value = "0", delval = "id")
    private Long deleted;
}

File: dehaze-go/internal/model/sys_ai_mcp_server.go
package model

import (
	"encoding/json"
	"time"
)

// SysAiMcpServer 外部 MCP Server 注册表
type SysAiMcpServer struct {
	BaseModel
	Name          string          `gorm:"column:name;type:varchar(128);not null;comment:Server名称(唯一)" json:"name"`
	Description   *string         `gorm:"column:description;type:varchar(512);comment:描述" json:"description"`
	ProtocolType  string          `gorm:"column:protocol_type;type:varchar(32);not null;default:streamable-http;comment:传输协议" json:"protocolType"`
	Endpoint      *string         `gorm:"column:endpoint;type:varchar(512);comment:端点URL" json:"endpoint"`
	AuthType      *string         `gorm:"column:auth_type;type:varchar(32);comment:鉴权方式" json:"authType"`
	Credentials   json.RawMessage `gorm:"column:credentials;type:json;comment:凭据密文(JSON)" json:"-"`
	Health        *string         `gorm:"column:health;type:varchar(16);comment:健康状态(online/offline)" json:"health"`
	LastCheckTime *time.Time      `gorm:"column:last_check_time;type:datetime;comment:最近一次健康探测时间" json:"lastCheckTime"`
	Status        int8            `gorm:"column:status;type:tinyint;not null;default:1;comment:状态(1:启用;0:禁用)" json:"status"`
	ToolCount     int             `gorm:"column:tool_count;type:int;not null;default:0;comment:工具数量" json:"toolCount"`
	Deleted       int64           `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAiMcpServer) TableName() string {
	return "sys_ai_mcp_server"
}
